docs/source/CartesianConfig.rst

```diff
@@ -363,6 +363,91 @@ statements addressing all objects. Note: This is contrary to the way the
 Cartesian configuration file as a whole is parsed (top-down).
 
 
+.. _heterogeneous_guest_variants:
+
+Heterogeneous guest variants
+============================
+
+The ‘``join``’ filter combined with the ‘``suffix``’ operator can be utilized
+together in order to produce guest variants with different guest OS or other
+types of configuration within the same test. Adding the ‘``suffix``’ keyword
+at the end of each variant to be combined and joining all the variants in the
+end will produce one final variant product with all participating variant
+dictionaries separately suffixed and combined into one final dictionary. For
+instance the following definition
+
+::
+
+    variants:
+        - one:
+            key1 = Hello
+            suffix _v1
+        - two:
+            key1 = Bye
+            key2 = World
+            suffix _v2
+        - three:
+    variants:
+        - four:
+            key3 = foo
+            only one
+        - five:
+            key3 = bar
+        - six:
+            key1 = foo
+            key2 = bar
+            only two
+
+    join one two
+
+will join variants one and two with suffixed parameters ``key1_v1 = Hello``
```

Comment:
> one and two

not quoted here as ``one`` and ``two``

Reply:
It would be great to mention there the case when keys and corresponding values have same value in both suffixed variant stanzas (e.g. ``one`` and ``two``). Resulting key in dict will not be suffixed then. I would incorporate this also in your example:
```
variants:
    - one:
        key1 = Hello
        key4 = Smart
        suffix _v1
    - two:
        key1 = Bye
        key2 = World
        key4 = Smart
        suffix _v2
```

``dict 1`` contains then:

```
dict    1:  four.one.five.two
    dep = []
    key1_v1 = Hello
    key1_v2 = Bye
    key2 = World
    key3 = bar
    key4 = Smart
    name = four.one.five.two
    shortname = four.one.five.two
```

Reply:
> > one and two
> 
> not quoted here as `one` and `two`

Good catch, fixed.

> It would be great to mention there the case when keys and corresponding values have same value in both suffixed variant stanzas

I was generally using a parameter defined in just one stanza for this purpose but you are right that it will be clearer to add one identical parameter to both stanzas in addition to the single-stanza parameter from before. Also fixed in the next push.